__log2 name conflicts with function in libm,
and so incorrect function is used.

diff -ur firefox-52.1.0esr/security/nss/lib/dbm/include/extern.h firefox-52.1.0esr/security/nss/lib/dbm/include/extern.h
--- firefox-52.1.0esr/security/nss/lib/dbm/include/extern.h	2017-05-04 12:06:18.612196876 +0300
+++ firefox-52.1.0esr/security/nss/lib/dbm/include/extern.h	2017-05-04 12:07:09.688943479 +0300
@@ -50,7 +50,7 @@
 BUFHEAD *__get_buf(HTAB *, uint32, BUFHEAD *, int);
 int __get_page(HTAB *, char *, uint32, int, int, int);
 int __ibitmap(HTAB *, int, int, int);
-uint32 __log2(uint32);
+uint32 __nss__log2(uint32);
 int __put_page(HTAB *, char *, uint32, int, int);
 void __reclaim_buf(HTAB *, BUFHEAD *);
 int __split_page(HTAB *, uint32, uint32);
diff -ur firefox-52.1.0esr/security/nss/lib/dbm/include/hash.h firefox-52.1.0esr/security/nss/lib/dbm/include/hash.h
--- firefox-52.1.0esr/security/nss/lib/dbm/include/hash.h	2017-05-04 12:06:18.612554454 +0300
+++ firefox-52.1.0esr/security/nss/lib/dbm/include/hash.h	2017-05-04 12:07:09.692663935 +0300
@@ -190,7 +190,7 @@
 #define OADDR_OF(S, O) ((uint32)((uint32)(S) << SPLITSHIFT) + (O))
 
 #define BUCKET_TO_PAGE(B) \
-    (B) + hashp->HDRPAGES + ((B) ? hashp->SPARES[__log2((uint32)((B) + 1)) - 1] : 0)
+    (B) + hashp->HDRPAGES + ((B) ? hashp->SPARES[__nss__log2((uint32)((B) + 1)) - 1] : 0)
 #define OADDR_TO_PAGE(B) \
     BUCKET_TO_PAGE((1 << SPLITNUM((B))) - 1) + OPAGENUM((B));
 
@@ -326,7 +326,7 @@
 BUFHEAD *__add_ovflpage(HTAB *hashp, BUFHEAD *bufp);
 int __big_insert(HTAB *hashp, BUFHEAD *bufp, const DBT *key, const DBT *val);
 int __expand_table(HTAB *hashp);
-uint32 __log2(uint32 num);
+uint32 __nss__log2(uint32 num);
 void __reclaim_buf(HTAB *hashp, BUFHEAD *bp);
 int __get_page(HTAB *hashp, char *p, uint32 bucket, int is_bucket, int is_disk, int is_bitmap);
 int __put_page(HTAB *hashp, char *p, uint32 bucket, int is_bucket, int is_bitmap);
diff -ur firefox-52.1.0esr/security/nss/lib/dbm/src/h_log2.c firefox-52.1.0esr/security/nss/lib/dbm/src/h_log2.c
--- firefox-52.1.0esr/security/nss/lib/dbm/src/h_log2.c	2017-05-04 12:06:18.614104502 +0300
+++ firefox-52.1.0esr/security/nss/lib/dbm/src/h_log2.c	2017-05-04 12:07:09.693421961 +0300
@@ -43,7 +43,7 @@
 #include "mcom_db.h"
 
 uint32
-__log2(uint32 num)
+__nss__log2(uint32 num)
 {
     register uint32 i, limit;
 
diff -ur firefox-52.1.0esr/security/nss/lib/dbm/src/hash.c firefox-52.1.0esr/security/nss/lib/dbm/src/hash.c
--- firefox-52.1.0esr/security/nss/lib/dbm/src/hash.c	2017-05-04 12:06:18.613698496 +0300
+++ firefox-52.1.0esr/security/nss/lib/dbm/src/hash.c	2017-05-04 12:07:09.706303625 +0300
@@ -353,13 +353,13 @@
         if (hashp->BSIZE > MAX_BSIZE)
             hashp->BSIZE = MAX_BSIZE;
 #endif
-        hashp->BSHIFT = __log2((uint32)hashp->BSIZE);
+        hashp->BSHIFT = __nss__log2((uint32)hashp->BSIZE);
     }
 
     if (info) {
         if (info->bsize) {
             /* Round pagesize up to power of 2 */
-            hashp->BSHIFT = __log2(info->bsize);
+            hashp->BSHIFT = __nss__log2(info->bsize);
             hashp->BSIZE = 1 << hashp->BSHIFT;
             if (hashp->BSIZE > MAX_BSIZE) {
                 errno = EINVAL;
@@ -406,7 +406,7 @@
      */
     nelem = (nelem - 1) / hashp->FFACTOR + 1;
 
-    l2 = __log2((uint32)PR_MAX(nelem, 2));
+    l2 = __nss__log2((uint32)PR_MAX(nelem, 2));
     nbuckets = 1 << l2;
 
     hashp->SPARES[l2] = l2 + 1;
@@ -425,7 +425,7 @@
                       1;
 
     nsegs = (nbuckets - 1) / hashp->SGSIZE + 1;
-    nsegs = 1 << __log2((uint32)nsegs);
+    nsegs = 1 << __nss__log2((uint32)nsegs);
 
     if (nsegs > hashp->DSIZE)
         hashp->DSIZE = nsegs;
@@ -1030,7 +1030,7 @@
      * * increases), we need to copy the current contents of the spare
      * split bucket to the next bucket.
      */
-    spare_ndx = __log2((uint32)(hashp->MAX_BUCKET + 1));
+    spare_ndx = __nss__log2((uint32)(hashp->MAX_BUCKET + 1));
     if (spare_ndx > hashp->OVFL_POINT) {
         hashp->SPARES[spare_ndx] = hashp->SPARES[hashp->OVFL_POINT];
         hashp->OVFL_POINT = spare_ndx;
